dot/network/config.go

```diff
@@ -97,6 +97,8 @@ type Config struct {
 
 	// telemetryInterval how often to send telemetry metrics
 	telemetryInterval time.Duration
+
+	noPreAllocate bool // internal option
```

Comment:
we want to pre-allocate by default so if this is changed to `preAllocate` then it will be false by default, but we won't know if it was set to false or if it defaulted to false, hence why it's `noPreAllocate`